### PR TITLE
Make `Pkg.activate(path)` behave like `pkg> activate path`.

### PR DESCRIPTION
**Pull request by fredrikekre**
Just moves the logic from REPLMode to API and make the REPL mode call API.

### CODE REVIEW

File: test/api.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
+module APITests
+
+using Pkg, Test
+
+@testset "Pkg.activate" begin
```
Comment (fredrikekre) [inline test/api.jl]: This testset exactly mirrors the one for activate in repl.jl, but with using the API. A bit unfortunate, would be nice to tests equivalent things like this in the same place

**Comment by KristofferC** [comment]
Yes this is good. Should be as little logic in the REPL functions as possible.

File: src/API.jl [hunk 2]
```diff
@@ -557,7 +557,30 @@ function status(ctx::Context, mode=PKGMODE_PROJECT)
 end
 
 function activate(path::Union{String,Nothing}=nothing)
-    Base.ACTIVE_PROJECT[] = Base.load_path_expand(path)
+    env = Base.active_project() === nothing ? nothing : EnvCache()
+    return activate(env, path)
+end
+function activate(env::Union{EnvCache,Nothing}, path::Union{String,Nothing})
```
Comment (fredrikekre) [inline src/API.jl]: Maybe we can call this `_activate`, or simply merge those two functions.

**Comment by fredrikekre** [comment]
Tests pass locally before the precompile default julia commit.

**Pull request merged**